pay-api/src/pay_api/models/routing_slip.py

```diff
@@ -151,9 +151,19 @@ def _add_date_filter(cls, query, search_filter):
 
     @classmethod
     def _add_receipt_number(cls, query, search_filter):
+        conditions = []
         if receipt_number := search_filter.get('receiptNumber', None):
-            query = query.join(Payment).filter(
-                and_(Payment.payment_account_id == PaymentAccount.id, Payment.cheque_receipt_number == receipt_number))
+            conditions.append(
+                and_(Payment.payment_account_id == PaymentAccount.id,
```

Comment:
may be this `                and_(Payment.payment_account_id == PaymentAccount.id,` can go on outer level if one of them presents, else may cause unwanted duplicate joins.